web-api/src/business/useCaseHelper/caseAssociation/updateCaseAndAssociations.ts

```diff
@@ -371,89 +407,11 @@ const updateCaseDeadlines = async ({
   });
 
   deadlines.forEach(caseDeadline => {
+    // @ts-ignore
     caseDeadline.associatedJudge = caseToUpdate.associatedJudge;
+    // @ts-ignore
     caseDeadline.associatedJudgeId = caseToUpdate.associatedJudgeId;
   });
   const validCaseDeadlines = CaseDeadline.validateRawCollection(deadlines);
-
-  return [() => upsertCaseDeadlines(validCaseDeadlines)];
-};
-
-/**
- * updateCaseAndAssociations
- * @param {object} providers the providers object
- * @param {object} providers.applicationContext the application context
- * @param {string} providers.caseToUpdate the case object which was updated
- * @returns {Promise<*>} the updated case entity
- */
-export const updateCaseAndAssociations = async ({
-  applicationContext,
-  authorizedUser,
-  caseToUpdate,
-  includeCorrespondenceAndWorkItems = true,
-}: {
-  applicationContext: ServerApplicationContext;
-  authorizedUser: UnknownAuthUser;
-  caseToUpdate: any;
-  includeCorrespondenceAndWorkItems?: boolean;
-}): Promise<RawCase> => {
-  const newCaseEntity: Case = caseToUpdate.validate
-    ? caseToUpdate
-    : new Case(caseToUpdate, {
-        authorizedUser,
-      });
-
-  const oldCaseEntity = await getCaseByDocketNumber({
-    applicationContext,
-    docketNumber: caseToUpdate.docketNumber,
-  });
-
-  const validNewRawCaseEntity = newCaseEntity.validate().toRawObject();
-
-  const validRawOldCaseEntity = new Case(oldCaseEntity, {
-    authorizedUser,
-  })
-    .validate()
-    .toRawObject();
-
-  let RELATED_CASE_OPERATIONS = [
-    updateCaseDeadlines,
-    updateCaseDocketEntries,
-    updateCaseMessages,
-    updateHearings,
-    updateIrsPractitioners,
-    updatePrivatePractitioners,
-  ];
-
-  if (includeCorrespondenceAndWorkItems) {
-    RELATED_CASE_OPERATIONS = RELATED_CASE_OPERATIONS.concat([
-      updateCaseWorkItems,
-      updateCorrespondence,
-    ]);
-  }
-
-  const validationRequests = RELATED_CASE_OPERATIONS.map(fn =>
-    fn({
-      applicationContext,
-      authorizedUser,
-      caseToUpdate: validNewRawCaseEntity,
-      oldCase: validRawOldCaseEntity,
-    }),
-  );
-
-  // wait for all validation tasks to complete and for callbacks to be generated
-  const persistenceCallbacks = (await Promise.all(validationRequests)).flat();
-
-  
-  // Persist primary case data first to ensure no errors
-  await upsertCases([validNewRawCaseEntity]);
-  
-  // Then persist related data
-  // all validation has passed, so now execute all persistence callbacks from results
-  const persistenceRequests = persistenceCallbacks.map(persistFn => {
```

Comment:
This is a bug. We should be returning the persistFn(). `persistenceRequests` is just an array of undefined.

Reply:
This has been the case for 4 years, based on git diff.